pkg/runtime/runtime.go

```diff
@@ -968,13 +968,14 @@ func (a *DaprRuntime) getTopicRoutes() (map[string]TopicRoute, error) {
 		return a.topicRoutes, nil
 	}
 
-	var topicRoutes map[string]TopicRoute = make(map[string]TopicRoute)
-
 	if a.appChannel == nil {
-		return topicRoutes, nil
+		return nil, errors.New("app channel not initialized, make sure -app-port is specified")
```

Comment:
See the original issue. There is a comment about this: https://github.com/dapr/dapr/issues/2979#issuecomment-807906198

A publisher does not need to expose a port, so this would artificially create a requirement.

Reply:
so if I don't misunderstand. We should log more instead of throw an error? If this is the case, I believe my change satisfy Yaron's comment, because

```go
func (a *DaprRuntime) startSubscribing() {
	for name, pubsub := range a.pubSubs {
		if err := a.beginPubSub(name, pubsub); err != nil {
			log.Errorf("error occurred while beginning pubsub %s: %s", name, err)
		}
	}
}
```

Reply:
@artursouza @halspang would you mind to shine more light on this?

Reply:
@artursouza in this case the subscription part will error out, this will not affect publishing. @beiwei30 is correct.

Reply:
I am actually concerned about logging this as an error. It is only an error if app intended to have a subscription but does not have app-port - we would need to "guess" the app's intent.

Reply:
Right. maybe change to warning? @beiwei30

Reply:
Any updates?

Reply:
noticed comments updated just now, will update the PR today.

Reply:
@yaron2 @artursouza 

I've updated the change like this:

```go
if a.appChannel == nil {
		log.Warn("app channel not initialized, make sure -app-port is specified if pubsub subscription is required")
		return topicRoutes, nil
	}
```

Pls. note there's no way to judge there are subscriptions expected on application side when the app port is not present.